src/databricks/labs/ucx/source_code/whitelist.py

```diff
@@ -0,0 +1,34 @@
+import sys
+from collections.abc import Iterable
+
+
+class Whitelist:
+    @classmethod
+    def parse(cls, data: str):
+        fixed_widths = [68, 9]  # name and ML-only
+        default_width = 18  # versions
+
+        def parse_line(line: str) -> list[str]:
+            cells: list[str] = []
+            for i in range(0, 1000):
+                count = fixed_widths[i] if i < len(fixed_widths) else default_width
+                cells.append(line[0:count])
+                line = line[count:]
+                if len(line) == 0:
+                    return cells
+            raise NotImplementedError()
+
+        lines = data.split("\n")
+        # skip header // header = parse_line(lines[0])
+        records = [parse_line(lines[i]) for i in range(1, len(lines))]
+        names = [record[0].strip() for record in records]
+        return Whitelist(names)
```

Comment:
You're assuming that the list of pre-installed packages is generated here, and meant for this use case only. It's not. So we're just reusing it, and extracting relevant info. Thanks.

Reply:
Plus we don't know yet if we'll only be using names, so may as well be ready.

Reply:
As per "YAGNI" principle, I think we just need to add whats necessary till the point of time. If we need ml and other cells, we can always add in later part which can also be verified using unit test.
@nfx

Reply:
why do we need a name len?...

Reply:
because the generated file follows a fixed length format

Reply:
i can re-generate a file in any format that is simple to consume :)

Reply:
Sure, but should you spend 15 minutes doing that, and should we spend 15 minutes discussing 5 lines of code that took me 3 minutes to write ? 
Maybe when we integrate this it will be a good time to reconsider this, since we will then know if we indeed only need names (and if it's worth the effort of changing it)?

Reply:
Closing since no longer relevant